jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -0,0 +1,196 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.logging;
+
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Consumer;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+
+public class JettyLoggerFactory implements ILoggerFactory
+{
+    private static final String ROOT_LOGGER_NAME = "";
+    private JettyLoggerConfiguration configuration;
+    private JettyLogger rootLogger;
+    private ConcurrentMap<String, JettyLogger> loggerMap;
+
+    public JettyLoggerFactory(JettyLoggerConfiguration config)
+    {
+        initialize(config);
+    }
+
+    /**
+     * Reinitialize this Factory, forgetting about all prior config and loggers.
+     * <p>
+     * This is really only used in test cases.
+     * </p>
+     *
+     * @param config the configuration to reset to
+     */
+    protected void initialize(JettyLoggerConfiguration config)
+    {
```

Comment:
```suggestion
```

Reply:
I need this for testing, all of your prior recommendations have eliminated all of my options, don't get rid of this one too.

Reply:
Done.